leasing/management/commands/load_csv_leasearea.py

```diff
@@ -0,0 +1,211 @@
+
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from leasing.enums import LeaseAreaType, LocationType
+from leasing.models.land_area import LeaseArea, LeaseAreaAddress
+from leasing.models.lease import Lease
+
+
+class Command(BaseCommand):
+    # Usage: `python manage.py load_csv_leasearea --csv path/to/file.csv`
+    help = "Load data from a CSV file to LeaseArea model and LeaseAreaAddress model"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--csv", type=str, required=False, help="The path to the CSV file"
+        )
+
+    def handle(self, *args, **kwargs):
+        csv_file = kwargs.get("csv")
+        if not csv_file:
+            # Print the expected headers if argument `--csv` is not provided
+            self.stdout.write(",".join(self._get_expected_headers()))
+            return
+
+        self.pre_validate_csv_rows(csv_file)
+
+        with open(csv_file) as file:
+            reader = csv.DictReader(
+                file,
+                delimiter=",",
+            )
+
+            try:
+                with transaction.atomic():
+                    for row in reader:
+                        self.create_leasearea(row)
+
+            except Exception as e:
+                self.stdout.write(self.style.ERROR(f"ERROR: Data loading failed: {e}"))
+
+        self.stdout.write(self.style.SUCCESS("Data loaded successfully"))
+
+    def _get_fieldnames_mappings(self):
+        fieldnames_mapping_lease = OrderedDict(
+            [
+                ("mvj_vuokraustunnus", "lease_identifier"),
+            ]
+        )
+        fieldnames_mapping_leasearea = OrderedDict(
+            [
+                ("mvj_kohteen_tunnus", "identifier"),
+                ("mvj_maaritelma", "type"),
+                ("mvj_pinta_ala", "area"),
+                ("mvj_sijainti", "location"),
+            ]
+        )
+        fieldnames_mapping_leasearea_address = OrderedDict(
+            [
+                ("mvj_osoite", "address"),
+                ("mvj_postinumero", "postal_code"),
+                ("mvj_kaupunki", "city"),
+                ("mvj_ensisijainen_osoite", "is_primary"),
+            ]
+        )
+
+        return (
+            fieldnames_mapping_lease,
+            fieldnames_mapping_leasearea,
+            fieldnames_mapping_leasearea_address,
+        )
+
+    def _transform_keys(self, row: dict, fieldnames_mapping: OrderedDict):
+        """Transform the keys in fieldnames_mapping from csv headers to model field names."""
+        return {
+            fieldnames_mapping[key]: value
+            for key, value in row.items()
+            if key in fieldnames_mapping
+        }
+
+    def _transform_type(self, value):
+        mapping = {
+            "Kaavayksikkö": LeaseAreaType.PLAN_UNIT,
+            "Kiinteistö": LeaseAreaType.REAL_PROPERTY,
+            "Määräala": LeaseAreaType.UNSEPARATED_PARCEL,
+            "Muu": LeaseAreaType.OTHER,
```

Comment:
If I understood the latest ticket comment correctly, Muu corresponds to PLAN_UNIT. No idea how the other types could be mapped then.

Reply:
Yep, need to probably go over all the possible options in detail and figure out what to do. It can be a bit misleading if one column has the same translation but maps to another one.

Reply:
Discussed in meeting: "Muu" is correct here, it's OTHER. The other values are not expected to be present in the input, but they don't do harm here anyway.